oml/models/vit/clip.py

```diff
@@ -0,0 +1,159 @@
+        ),
+        "openai_vitl14_336": (
+            "https://openaipublic.azureedge.net/clip/models/3035c92b350959924f9f00213499208652fc7ea050643e8b385c2dac08641f02/ViT-L-14-336px.pt",
+            "b311058cae50cb10fbfa2a44231c9473",
+        ),
+        # checkpoints pretrained by SberbankAI
+        "sber_vitb16_224": (
+            "https://huggingface.co/sberbank-ai/ruclip-vit-base-patch16-224/resolve/main/pytorch_model.bin",
+            "7882e07674d78c674e33cb892a68bbfc",
+        ),
+        "sber_vitb32_224": (
+            "https://huggingface.co/sberbank-ai/ruclip-vit-base-patch32-224/resolve/main/pytorch_model.bin",
+            "e2c4dab46a3cfa608bdd762973e90d32",
+        ),
+        "sber_vitl14_224": (
+            "https://huggingface.co/sberbank-ai/ruclip-vit-large-patch14-224/resolve/main/pytorch_model.bin",
+            "9b4a1cd25d15bad4ffd2ba6e34b8a67c",
+        ),
+    }
+
+    def __init__(
+        self,
+        arch: str,
+        normalise_features: bool = True,
+        weights: Optional[str] = None,
+        strict_load: bool = True,
+    ):
+        """
+        Args:
+            weights: Path to weights or special key for pretrained ones or ``None`` for random initialization.
+             You can check available pretrained checkpoints in ``ViTCLIPExtractor.pretrained_models``.
+            arch: Might be one of ``vitb16_224``, ``vitb32_224``, ``vitl14_224``, ``vitl14_336``.
+            normalise_features: Set ``True`` to normalise output features
+            strict_load: Whether the weights needed to be loaded strictly. Doesn't work with OpenAI's models.
+        """
+
+        super().__init__()
+
+        self.normalize = normalise_features
+
+        cfg = get_vit_config_by_name(arch)
+        embed_dim = cfg["embed_dim"]
+        image_resolution = cfg["image_resolution"]
+        layers = cfg["layers"]
+        width = cfg["width"]
+        patch_size = cfg["patch_size"]
+        heads = cfg["heads"]
+
+        self.visual = VisionTransformer(
+            input_resolution=image_resolution,
+            patch_size=patch_size,
+            layers=layers,
+            width=width,
+            heads=heads,
+            output_dim=embed_dim,
+        )
+
+        if weights is None:
+            return
+        if weights in self.pretrained_models:
+            weights, md5 = self.pretrained_models[weights]
+            weights = download_checkpoint(weights, md5)
+
+        try:  # check if weights are jitted
+            visual = torch.jit.load(Path(weights), map_location="cpu").visual.eval()
+            patch_device_and_float(visual, device="cpu")
+            state_dict = visual.state_dict()
+        except RuntimeError:
+            state_dict = torch.load(Path(weights), map_location="cpu")
+            state_dict = state_dict.get("state_dict", state_dict)
+            state_dict = remove_prefix_from_state_dict(state_dict, trial_key="conv1.weight", skip_unnecessary=True)
```

Comment:
I was not sure about the need of this argument - `skip_unnecessary`
probably, after the refactoring suggested above we ca get rid of it